plugins/bit_invader/bit_invader.cpp

```diff
@@ -65,17 +65,16 @@ Plugin::Descriptor PLUGIN_EXPORT bitinvader_plugin_descriptor =
 }
 
 
-bSynth::bSynth( float * _shape, int _length, NotePlayHandle * _nph, bool _interpolation,
+bSynth::bSynth( float * _shape, NotePlayHandle * _nph, bool _interpolation,
 				float _factor, const sample_rate_t _sample_rate ) :
 	sample_index( 0 ),
 	sample_realindex( 0 ),
 	nph( _nph ),
-	sample_length( _length ),
 	sample_rate( _sample_rate ),
 	interpolation( _interpolation)
 {
-	sample_shape = new float[sample_length];
-	for (int i=0; i < _length; ++i)
+	sample_shape = new float[200];
+	for (int i=0; i < 200; ++i)
```

Comment:
The `_shape` parameter contains the values from the `bitInvader`'s `m_graphModel`, but this only has an initial size of 128. The new code here could read memory out-of-bounds until the graph's length is increased to 200 for the first time. You'll want to ensure that the graph has the full 200 samples when the `bitInvader` is constructed.

Reply:
My OCD tells me that we should increase it to 256. Cause that's what the length would be on an old wavetable synt.h. Or a new one (http://www.doepfer.de/a112.htm).

Reply:
I agree.  The maximum being at 200 and not 256 has always bugged me.